### ISSUE TITLE
Figure out torch placing logic

### ISSUE THREAD

**Issue by tonihele** [tonihele/OpenKeeper#128]
Figure out torch placing logic. Maybe it is simple, maybe it is Maybelline... They are never adjacent to each other, max one per tile. Is it predetermined per tile? Which side of a corridor to place each one.

**Comment by ArchDemons** [comment]
Torch placing logic is very simple.
- Torch placed in tiles with odd coordinates (x, y). E.g Point(1, 3), Point(5, 7)
- Only one torch available in tile
- Torch attach to first neighbor tile wall which contain flag TORCH.

**Comment by tonihele** [comment]
Great! Odd coordinates? Ours or theirs? We have 0-based, and at least the map editor has 1-based. We could try placing the torches. Add real lights in debug mode etc. And try the scene graph options in JME 3.1 branch (which I try to create soonish).

**Comment by ArchDemons** [comment]
odd with our 0-based coordinate system

**Comment by tonihele** [comment]
I can try to place these, we need object loader possibly with this. At least initial implementation. #20. Let's close this when we are sure.

**Comment by tonihele** [comment]
Hmm, only one per tile. But all tiles can contain torch. Even or odd. It has to do something with wall placing logic. No two adjacent walls can have torches.

**Comment by tonihele** [comment]
![torches](https://cloud.githubusercontent.com/assets/8344766/13394686/badb176c-def2-11e5-80a7-496a869ac1a1.png)

**Comment by tonihele** [comment]
I committed simple logic. Just at least to allow us to play with effects & lights.

**Comment by ArchDemons** [comment]
Some correcting my algorithm. X - axis place only odd x-coordinate on SOUTH and NORTH walls. Y-axis place only odd y-coordinate on WEST and EAST walls (something like this)


![torch_2](https://cloud.githubusercontent.com/assets/12515916/13397755/63380b68-df0c-11e5-97e6-e1be0f0c1ea9.png)

**Comment by ufdada** [comment]
i don't think it works like that. Look here:
![shot0003](https://cloud.githubusercontent.com/assets/3787188/13404658/08902858-df1b-11e5-8a0e-d66fd5596d97.jpg)

**Comment by ArchDemons** [comment]
![unnamed](https://cloud.githubusercontent.com/assets/12515916/13409715/3d413aae-df45-11e5-9a14-4a2f62f006e5.jpg)

**Comment by ufdada** [comment]
We get there :), but the tile left to the smaller circle should have a torch which it doens't have

**Comment by ArchDemons** [comment]
It should not. More specifically can, but not on this side of the wall (only on walls with blue lines)

**Comment by Trass3r** [comment]
The current code doesn't work. With a simple test map I get no torches at all.
[testmap.zip](https://github.com/tonihele/OpenKeeper/files/13055821/testmap.zip)

**Comment by Trass3r** [comment]
https://discord.com/channels/480505152806191114/1269093750551089253/1280037150007492676

**Comment by belgianguy** [comment]
<img width="2240" height="768" alt="Image" src="https://github.com/user-attachments/assets/bfbbc204-3ee3-45dd-a98d-3f892b7e9906" />

**Comment by belgianguy** [comment]
Good catch. With the hard-coded assignment, `roomInstance` was indeed always `null`, making the room-specific torch logic unreachable.

Simply restoring:
```java
roomCoordinates.get(tile.getLocation())
```
would still be unreliable. `handleTorch()` runs before `handleRoom()` for ordinary rooms, so the first processed room tile may not yet have a registered `RoomInstance`.

I changed the code to resolve the room definition directly from the tile terrain:
```java
Room room = kwdFile.getRoomByTerrain(tile.getTerrainId());
ArtResource torch = room.getTorch();
```
This avoids the initialization-order dependency. A null torch resource now suppresses generic room-wall torches; otherwise, the configured model is used.

### Room torch definitions

I inspected the room definitions across all 72 supplied DKII maps. Every non-null room torch field selects the same model: `Torch1`. There are no alternative room-specific torch models in those maps.

Rooms configured with `Torch1`:

- Casino
- Combat Pit
- Crypt
- Dungeon Heart
- Graveyard
- Guard Room
- Hatchery
- Hero Gate [2x2]
- Hero Stone Bridge
- Lair
- Library
- Portal
- Prison
- Stone Bridge
- Torture Chamber
- Training Room
- Treasury
- Wooden Bridge
- Workshop

Rooms with no torch configured:

- Temple
- Hero Gate [Final]
- Hero Gate [Tile] [not used]
- Hero Gate [FrontEnd]
- Hero Gate [3x1]
- Mercenary Gate [not used]
- Hero Portal

Level 1 uses `Hero Gate [2x2]`, so the torches visible around that gate are expected: this gate variant explicitly selects `Torch1`. Other Hero Gate variants do not.

### Temple investigation

We found an interesting live-update bug in original DKII:

1. Start with reinforced walls that already have torches.
2. Build a Temple against those walls.
3. The pre-existing torches incorrectly remain during live play.
4. Save and reload the game.
5. The wall torches disappear.

This indicates that building a Temple does not remove stale wall decorations immediately. A full load reconstructs the scene and correctly respects the Temple’s null torch resource.

The Temple’s four large glowing corner fixtures are separate parts of the Temple construction and should not be confused with the generic wall-mounted `Torch1` model.

### OpenKeeper verification

To test the exact case, we generated `TempleTorchWalls.kwd` containing:

- An exact 5×5 Temple
- A complete 24-tile ring of player-owned Reinforced Wall
- A solid exterior ring, ensuring that visible inner-face torches could only originate from the Temple side

With the Temple present, OpenKeeper renders no generic `Torch1` flames on the surrounding reinforced walls. Only the Temple’s four built-in corner lights remain.

After selling the Temple, its tiles become claimed path and generic `Torch1` decorations immediately appear on the same inner wall faces according to the normal coordinate pattern.

Therefore, OpenKeeper now consistently matches DKII’s reconstructed/post-load state while avoiding both the native stale-decoration bug and the `RoomInstance` initialization-order problem.
```
